pkg/operations/helm3/helm3.go

```diff
@@ -32,14 +35,32 @@ import (
 
 const (
 	defaultRootPath = "resources"
+
+	errEmptyParentResource = "parent resource is empty"
+	errSpecCast            = "parent resource spec could not be casted into a map[string]interface{}"
+	errParse               = "could not parse the generated YAMLs"
 )
 
+// WithResourcePath returns an Option that changes the resource path of the Engine.
 func WithResourcePath(path string) Option {
 	return func(h *Engine) {
 		h.ResourcePath = path
 	}
 }
 
+// WithLogger returns an Option that changes the logger of the Engine.
+func WithLogger(l logging.Logger) Option {
+	return func(h *Engine) {
+		// NOTE(muvaf): Even though l.Debug seems to satisfy action.DebugLog interface,
+		// they are completely different given that former user the first argument
+		// as context while the latter uses it as format string.
+		h.log = func(format string, v ...interface{}) {
+			l.Debug(fmt.Sprintf(format, v...))
```

Comment:
It seems like we cast everything to debug level here. What's the reason we need to do that?

Reply:
Yes, we pass this logger down to Helm3's `Init` function which has the following signature:
```
Init(getter genericclioptions.RESTClientGetter, namespace string, helmDriver string, log DebugLog) error
```

As you see, its logger input is only debug level. In direct function calls made by us, error is returned and we are expected to handle the logging as we want.